src/Entities/Misc/DreamStrawberry.cs

```diff
@@ -0,0 +1,130 @@
+
+            // Removes any default DashListeners from the strawberry as we do not want to use those
+            foreach (Component comp in Components.ToArray()) {
+                if (comp is DashListener)
+                    Components.Remove(comp);
+            }
+
+            // To account for the DashListeners I just brutally murdered, we add a DreamDashListener instead
+            // as we want it to activate from dream blocks and not normal player dashes
+            Add(new DreamDashListener {
+                OnDreamDash = new Action<Vector2>(OnDreamBerryDash)
+            });
+
+            dreamStrawberryData = DynamicData.For(this);
+        }
+
+        #region Bad Code
+        /*
+        public override void Added(Scene scene) {
+            sprite.OnFrameChange = OnAnimate;
+            Add(wiggler = Wiggler.Create(0.4f, 4f, delegate (float v)
+            {
+                sprite.Scale = Vector2.One * (1f + v * 0.35f);
+            }));
+            Add(rotateWiggler = Wiggler.Create(0.5f, 4f, delegate (float v)
+            {
+                sprite.Rotation = v * 30f * ((float) Math.PI / 180f);
+            }));
+            Add(bloom = new BloomPoint(1f, 12f));
+            Add(light = new VertexLight(Color.White, 1f, 16, 24));
+            Add(lightTween = light.CreatePulseTween());
+            if ((scene as Level).Session.BloomBaseAdd > 0.1f) {
+                bloom.Alpha *= 0.5f;
+            }
+        }
+
+        public void OnAnimate(string id) {
+            if (!flyingAway && id == "flap" && sprite.CurrentAnimationFrame % 9 == 4) {
+                Audio.Play("event:/game/general/strawberry_wingflap", Position);
+                flapSpeed = -50f;
+            }
+            int num = 25;
+            // Checks if the animation from is 25, runs the funny strawberry noises
+            if (sprite.CurrentAnimationFrame == num) {
+                lightTween.Start();
+                if (!collected && (CollideCheck<FakeWall>() || CollideCheck<Solid>())) {
+                    Audio.Play("event:/game/general/strawberry_pulse", Position);
+                    SceneAs<Level>().Displacement.AddBurst(Position, 0.6f, 4f, 28f, 0.1f);
+                } else {
+                    Audio.Play("event:/game/general/strawberry_pulse", Position);
+                    SceneAs<Level>().Displacement.AddBurst(Position, 0.6f, 4f, 28f, 0.2f);
+                }
+            }
+        }
+        */
+        #endregion
+
+        public override void Update() {
+
+            base.Update();
+
+            // Creates and updates the dream trail
+            if (Scene.OnInterval(0.1f))
+                CreateDreamTrail();
+        }
+
+        // Calls the original OnDash from our DreamDashListener
+        public void OnDreamBerryDash(Vector2 dir) {
+            m_Strawberry_OnDash.Invoke(this, new object[] { dir });
+        }
```

Comment:
I wonder if this could be handled with a `MonoModLinkTo`? something like 
```suggestion
        // Calls the original OnDash from our DreamDashListener
        [MonoModLinkTo("Celeste.Strawberry", "OnDash(Microsoft.Xna.Framework.Vector2)")]
        public extern void OnDreamBerryDash(Vector2 dir);
```
no clue if that actually works but it'd be a really clean way to do this if it does

Reply:
well yeah in theory this would be a clean solution, but this implementation causes the berry to refuse to load, so i think the current implementation is good enough for now.

i dont really see a need to replace it either way